noxfile.py

```diff
@@ -86,11 +88,25 @@ def build(session):
     session.run("python", "setup.py", "--quiet", "sdist", "bdist_wheel")
 
 
+def has_changes():
+    status = (
+        subprocess.run("git status --porcelain", shell=True, stdout=subprocess.PIPE)
+        .stdout.decode()
+        .strip()
+    )
+    return len(status) > 0
```

Comment:
Maybe check the status code well? Otherwise it’d be reporting no changes when Git fails without output.